transaction-impl/src/main/java/Module.java

```diff
@@ -12,7 +18,15 @@
     @Override
     protected void configure() {
         bindService(TransactionService.class, TransactionServiceImpl.class);
-
         bindClient(ItemService.class);
+
+        bind(TransactionServiceLifecycle.class).asEagerSingleton();
+    }
+
+    private static class TransactionServiceLifecycle {
+        @Inject
+        public TransactionServiceLifecycle(ApplicationLifecycle applicationLifecycle, PersistentEntityRegistry persistentEntityRegistry) {
+            applicationLifecycle.addStopHook(() -> persistentEntityRegistry.gracefulShutdown(FiniteDuration.create(5, TimeUnit.SECONDS)));
+        }
```

Comment:
That's a great approach. We usually add the `addStopHook` in the constructor of the `XxxServiceImpl` but this pattern makes things cleaner IMHO (keeps all wiring and shutdown in a single place).